oauth/oauth.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:flutter_appauth/flutter_appauth.dart';
+
+class AuthRepository {
+  AuthRepository();
+
+  final _auth = const FlutterAppAuth();
+
+  @override
+  Future<Authorization?> authorize() async {
+    try {
+      final response = _auth.authorizeAndExchangeCode(
+        AuthorizationTokenRequest(
+          '<client_id>',
+          '<redirect_url>',
+          issuer: '<issuer>',
+          clientSecret: '<client_secret>',
+          scopes: ['openid', 'profile', 'email', 'offline_access', 'api'],
+          preferEphemeralSession: true,
+        ),
+      );
+      return Authorization(
+        refreshToken: response.refreshToken!,
+        accessToken: response.accessToken!,
+      );
+    } catch (_) {
+      return null; // exception is thrown if user cancels authorization
+    }
+  }
+
+  @override
+  Future<Authorization?> refreshToken(String refreshToken) async {
```

Comment:
Do we use this in some intercepter?
If so can we add that interceptor to here as well?

Reply:
This is used in the `AuthorizationInterceptor` to refresh the expired token.
I would avoid adding the `AutorizationInterceptor` to this chapter, as this chapter should only teach how to use OAuth.
The `AutorizationInterceptor` is quite complex. Refreshing the token is only a small part of what it does.

Reply:
I added the `AuthorizationInterceptor`.